features/sharding/core/src/main/java/org/apache/shardingsphere/sharding/yaml/engine/representer/processor/NoneYamlTupleProcessor.java

```diff
@@ -37,11 +37,11 @@ public String getTupleName() {
     
     @Override
     public NodeTuple process(final NodeTuple nodeTuple) {
-        return (isNullNode(nodeTuple.getValueNode()) || isEmptyMappingNode(nodeTuple.getValueNode())) ? null : processNoneTuple(nodeTuple);
+        return (isNullNode(nodeTuple.getKeyNode(), nodeTuple.getValueNode()) || isEmptyMappingNode(nodeTuple.getValueNode())) ? processNoneTuple(nodeTuple) : null;
     }
     
-    private boolean isNullNode(final Node valueNode) {
-        return Tag.NULL.equals(valueNode.getTag());
+    private boolean isNullNode(final Node keyNode, final Node valueNode) {
+        return !"none".equalsIgnoreCase(((ScalarNode) keyNode).getValue()) && Tag.NULL.equals(valueNode.getTag());
```

Comment:
If yaml text contains `none: 'xxx'`, then node is handled as null. Could we add more accurate check here?

Reply:
If yaml text contains `none: 'xxx'` , the node will still be `none`
![image](https://user-images.githubusercontent.com/57847965/218402232-3b2d0643-50bd-48eb-9f7c-503e3ad0b0d3.png)

![image](https://user-images.githubusercontent.com/57847965/218402173-fde6cb1a-a83b-463a-b3e1-30147ede0310.png)

Reply:
I removed this change cuz it's useless

Reply:
OK